### PR TITLE
Highlight cache usage in resin build

### PR DESCRIPTION
**Pull request by CameronDiver**
This commit will highlight the usage of the cache when doing a docker
build via `resin build`, which not only helps the user understand what
the build is doing, but also achieves more parity with the cloud
builder.

Change-type: patch
Signed-off-by: Cameron Diver <cameron@resin.io>

### CODE REVIEW

File: lib/utils/docker.coffee [hunk 1]
```diff
@@ -110,13 +110,34 @@ exports.tarDirectory = tarDirectory = (dir) ->
...
+		arrowTest = /^\s*-+>\s*(.+)/i
+		if (match = arrowTest.exec(message))
+			match[1]
+		else
+			false
+
+	es.map (data, cb) ->
+		msg = extractArrowMessage(data)
+		if msg and msg.toLowerCase() == 'using cache'
+			data = colors.bgGreen.black(msg)
```
Comment (pimterry) [inline lib/utils/docker.coffee]: Playing around with `colors`, it looks like if `msg` already has a color this won't work. Are we sure it always doesn't?

Comment (CameronDiver) [inline lib/utils/docker.coffee]: I'm fairly sure it won't, unless Docker changes it's output format, but in that case other changes will be needed anyway.

File: lib/utils/docker.coffee [hunk 2]
```diff
@@ -110,13 +110,33 @@ exports.tarDirectory = tarDirectory = (dir) ->
...
+	{ EOL } = require('os')
+
+	extractArrowMessage = (message) ->
+		arrowTest = /^\s*-+>\s*(.+)/i
+		if (match = arrowTest.exec(message))
+			match[1]
+		else
+			undefined
+
+	es.map (data, cb) ->
```
Comment (Page-) [inline lib/utils/docker.coffee]: `es.mapSync` would be nicer:
```coffee
es.mapSync (data, cb) ->
	msg = extractArrowMessage(data)
	if msg? and msg.toLowerCase() == 'using cache'
		data = colors.bgGreen.black(msg)
	return data + EOL

File: lib/utils/docker.coffee [hunk 3]
```diff
@@ -110,13 +110,33 @@ exports.tarDirectory = tarDirectory = (dir) ->
 		pack.finalize()
 		return pack
 
+cacheHighlightStream = ->
+	colors = require('colors')
```
Comment (Page-) [inline lib/utils/docker.coffee]: Can you use `colors/safe` to avoid messing with the string prototype (you already use the "safe" api below)

**Pull request merged**